src/tr1/game/objects/effects/body_part.c

```diff
@@ -58,7 +61,9 @@ void BodyPart_Control(int16_t fx_num)
             fx->speed = 0;
             fx->frame_num = 0;
             fx->counter = 0;
-            fx->object_id = O_EXPLOSION_1;
+            if (Mutant_EnableExplosion()) {
+                fx->object_id = O_EXPLOSION_1;
+            }
```

Comment:
I'm not positive this check is necessary. 🤔

Reply:
If we don't assing fx->object_id, the game should crash not knowing what object is tied to the created effect.

Reply:
The game doesn't crash for me with this addition. And the behavior matches with the `Testing chart` above. Maybe @lahm86 has an idea.

I understand wanting to match the OG behavior, but it makes me confused about why we inject the explosion sprite if we don't want to use it in non-OG cases.

Reply:
To clarify, any FX that doesn't have its `object_id` set is considered not valid (see `Effect_Control`, `Effect_Draw` which use this property unconditionally). Not setting it also explains why the explosion sprites are not working with `/abortion`, which is the main reason why we inject the sprites to begin with.

Reply:
Ideally `BodyPart_Control` would know which object it's related to, or have a "do explode" flag passed to it from `Effect_ExplodingDeath` (which knows about the item). But that would mean storing it somewhere - maybe it could be packed into another field?

Reply:
Alright – I see what's going on – it converts the body part FX into an explosion, rather than creating a new `FX` for it. Forget about my earlier comments.

Upon examining the code closer, the explosions can already be turned on or off by making the damage passed from `Effect_ExplodingDeath` positive or negative, respectively. So for the overall fix, it's sufficient to replace `FLYER_SMARTNESS` with `FLYER_PART_DAMAGE` which was likely a typo.